cypress/integration/SelectLabs.spec.ts

```diff
@@ -0,0 +1,190 @@
+              h.selectLabs.getListbox().should('be.focused');
+            });
+
+            it('should have an aria-activedescendant attribute with the same value as the id of the first option', () => {
+              h.selectLabs.getListbox().should($listbox => {
+                h.selectLabs.assertADMatchesOption($listbox, 0);
+              });
+            });
+          });
+
+          context('when the down key is pressed for a second time', () => {
+            beforeEach(() => {
+              h.selectLabs.getListbox().type('{downarrow}');
+            });
+
+            context('the listbox', () => {
+              it('should have an aria-activedescendant attribute with the same value as the id of the second option', () => {
+                h.selectLabs.getListbox().should($listbox => {
+                  h.selectLabs.assertADMatchesOption($listbox, 1);
+                });
+              });
+            });
+
+            context('when the down key is pressed for a third time', () => {
+              beforeEach(() => {
+                h.selectLabs.getListbox().type('{downarrow}');
+              });
+
+              context('the listbox', () => {
+                it('should have an aria-activedescendant attribute with the same value as the id of the fourth option (having skipped the third option since it was disabled)', () => {
+                  h.selectLabs.getListbox().should($listbox => {
+                    h.selectLabs.assertADMatchesOption($listbox, 3);
+                  });
+                });
+              });
+
+              context('when the enter key is pressed', () => {
+                beforeEach(() => {
+                  h.selectLabs.getListbox().type('{enter}');
+                });
+
+                context('the select button', () => {
+                  it(`should read "${enteredLabel}"`, () => {
+                    h.selectLabs.getButton().should('have.text', enteredLabel);
+                  });
+
+                  it(`should re-acquire focus`, () => {
+                    h.selectLabs.getButton().should('be.focused');
+                  });
+                });
+
+                context('the hidden text input', () => {
+                  it(`should have a value of "${enteredValue}"`, () => {
+                    h.selectLabs.getInput().should('have.value', enteredValue);
+                  });
+                });
+
+                context('the listbox', () => {
+                  it('should be dismissed', () => {
+                    h.selectLabs.getListbox().should('not.be.visible');
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  // TODO: Re-enable these tests
+  /*
+  context(`given the 'Disabled' story is rendered`, () => {
+    beforeEach(() => {
+      h.stories.load('Components|Inputs/Select/React/Top Label', 'Disabled');
+    });
+
+    it('should pass accessibility checks', () => {
```

Comment:
We should be more clear about what is being checked in `checkA11y`. It isn't really "accessibility checks", it is "axe-core" checks. Axe core doesn't catch all or even most accessibility issues. Our A11y team figures it is less than 20%. That's why we have manually specified accessibility specifications. We figure axe-core checks are easy enough to just throw in there just in case we missed something, but the output is harder to understand and know what action needs to be taken.

Reply:
Yep, I remember you mentioning this earlier. I'll update the wording.

Reply:
Fixed